packages/node_modules/@webex/internal-plugin-calendar/src/calendar.js

```diff
@@ -2,10 +2,225 @@
+   * @event calendar:meeting:create
+   * @instance
+   * @memberof Calendar
+   */
+
+/**
+   * Calendar Item Update Event
+   * Emitted when a calendar item has been updated
+   * @event calendar:meeting:update
+   * @instance
+   * @memberof Calendar
+   */
+
+/**
+   * Calendar Item Update Event
+   * Emitted when a calendar item has been deleted
+   * @event calendar:meeting:delete
+   * @instance
+   * @memberof Calendar
+   */
+
+/**
+ * Calendar Registered Event
+ * Emitted when the calendar instance has been registered and listening
+ * @event calendar:registered
+ * @instance
+ * @memberof Calendar
+ */
+
+/**
+ * Calendar Registered Event
+ * Emitted when the calendar instance has been registered and listening
+ * @event calendar:unregistered
+ * @instance
+ * @memberof Calendar
+ */
+
 import {WebexPlugin} from '@webex/webex-core';
 
+import CalendarCollection from './collection';
+
 const Calendar = WebexPlugin.extend({
   namespace: 'Calendar',
+  registered: false,
+
+  /**
+     * Explicitly sets up the calendar plugin by registering
+     * the device, connecting to mercury, and listening for calendar events.
+     * @returns {Promise}
+     * @public
+     * @memberof Calendar
+     */
+  register() {
+    if (!this.webex.canAuthorize) {
+      this.logger.error('calendar->register#ERROR, Unable to register, SDK cannot authorize');
+
+      return Promise.reject(new Error('SDK cannot authorize'));
+    }
+
+    if (this.registered) {
+      this.logger.info('calendar->register#INFO, Meetings plugin already registered');
+
+      return Promise.resolve();
+    }
+
+    return this.webex.internal.device.register()
+      .then(() => this.webex.internal.mercury.connect())
+      .then(() => {
+        this.listenForEvents();
+        this.trigger('calendar:registered');
+        this.registered = true;
+      })
+      .catch((error) => {
+        this.logger.error(`calendar->register#ERROR, Unable to register, ${error.message}`);
+
+        return Promise.reject(error);
+      });
+  },
+
+  /**
+     * Explicitly tears down the calendar plugin by deregistering
+     * the device, disconnecting from mercury, and stops listening to calendar events
+     *
+     * @returns {Promise}
+     * @public
+     * @memberof Meetings
+     */
+  unregister() {
+    if (!this.registered) {
+      this.logger.info('meetings->unregister#INFO, Meetings plugin already unregistered');
+
+      return Promise.resolve();
+    }
+
+    this.stopListeningForEvents();
+
+    return this.webex.internal.mercury.disconnect()
+      .then(() => this.webex.internal.device.unregister())
+      .then(() => {
+        this.trigger('calendar:unregistered');
+        this.registered = false;
+      });
+  },
+
+  /**
+   * registers for calendar events through mercury
+   * @returns {undefined}
+   */
+  listenForEvents() {
+    // Calendar mercury events listener
+    this.webex.internal.mercury.on('event:calendar.meeting.create', (envelope) => {
```

Comment:
Can we make these event string constants like we have them in the meetings plugin as `CALENDAR_EVENTS`?